app/components/Editor.js

```diff
@@ -129,23 +132,39 @@ export default class Editor extends React.Component {
         $R('#editor', 'module.buffer.clear');
       },
       setEditorRawText: (rawText) => {
-        // Get the current caret location
-        let caretLocation = this.redactor.selection.getPosition();
-        let point = {clientX: caretLocation.left, clientY: caretLocation.top};
-
-        let cleaned = this.redactor.cleaner.input(rawText);
+        // Sets the code for the editor, maintains caret location if available.
+
+        let focused = this.redactor.editor.isFocus();
+        let point;
+
+        if (focused) {
+          // Attempt to save caret location, otherwise ignore.
+          try {
```

Comment:
Do we know in what circumstances it would fail? A try/catch can plug the problem, but doesn't help us understand the problem in general.

Reply:
It will fail whenever the editor is not in focus. But in some situations the editor will focus on load but there is nothing in the editor, I think it will also fail.

I'll push another commit which updates things a bit more.